src/GitVersionCore/FileLocking/FileLockContextExtensions.cs

```diff
@@ -0,0 +1,38 @@
+/*
+MIT License
+
+Copyright 2020 Teroneko
+
+Permission is hereby granted, free of charge, to any person obtaining a copy 
+of this software and associated documentation files (the "Software"), to deal 
+in the Software without restriction, including without limitation the rights to 
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies 
+of the Software, and to permit persons to whom the Software is furnished to do 
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all 
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+namespace GitVersion.FileLocking
+{
+
+#nullable enable
+
+    internal static class FileLockContextExtensions
+    {
+        public static bool IsErroneous(this FileLockContext? fileLockContext)
```

Comment:
Why isn't this just an instance method on the `FileLockContext` class?

Reply:
Because the parameter "fileLockContext" can be null. With this "trick" you can shift the null check to this method. That's simply a shortcut.